script.js

```diff
@@ -1,12 +1,177 @@
+        </div>
+      </div>
+      <div class="col-md-3">
+        <div class="card-body">
+          <p class="card-text"><strong>Rating:</strong> ${obj.rating.average}</p>
+          <p class="card-text"><strong>Genre:</strong> ${obj.genres}</p>
+          <p class="card-text"><strong>Status:</strong> ${obj.status}</p>
+          <p class="card-text"><strong>Runtime:</strong> ${obj.runtime}</p>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>`;
+
+function fetchAllShowsAndDisplay() {
+  fetch('https://api.tvmaze.com/shows')
+    .then(res => res.json())
+    .then(data => {
+      allTVShows = data;
+      displayAllShows(allTVShows);
+      getAllGenres();
+    })
+    .catch(error => console.log(error))
+}
+
+fetchAllShowsAndDisplay();
+
+function fetchShowAndDisplay(id) {
+  fetch(`https://api.tvmaze.com/shows/${id}/episodes`)
+    .then(res => res.json())
+    .then(data => {
+      allEpisodes = data;
+      displaySelectedShow(allEpisodes);
+    })
+    .catch(error => console.log(error));
+}
+
+
+function displayEpisodes(arr) {
+  showsEl.setAttribute('style', 'display:none !important');
+  episodesEl.setAttribute('style', 'display:inline');
+  let allEpisodesHTML = '';
+  arr.forEach(episode => allEpisodesHTML += getEpisodeCard(episode));
+  episodesEl.innerHTML = allEpisodesHTML;
+  searchResult.innerHTML = `Displaying ${arr.length}/${arr.length}`;
+  searchShowsEl.setAttribute('style', 'display: none !important');
+  searchEpisodesEl.setAttribute('style', 'display: inline');
+}
+
+function createEpisodeSelect(arr) {
+  selectEpisodesEl.innerHTML = '';
+  let selectAllEpisodes = document.createElement('option');
+  selectAllEpisodes.innerText = 'All Episodes';
+  selectAllEpisodes.value = 'selectAll';
+  selectEpisodesEl.appendChild(selectAllEpisodes);
+  arr.forEach(episode => {
+    let selectOption = document.createElement('option');
+    selectOption.innerText = `${getEpisodeTitle(episode)}`;
+    selectOption.value = episode.id;
+    selectEpisodesEl.appendChild(selectOption);
+  });
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+function displaySelectedShow(arr) {
+  createEpisodeSelect(arr);
+  displayEpisodes(arr);
+  searchEpisodesEl.value = '';
+  searchShowsEl.setAttribute('style', 'display: none !important');
+  searchEpisodesEl.setAttribute('style', 'display: inline');
 }
 
-window.onload = setup;
+selectEpisodesEl.addEventListener('change', (e) => {
+  if (e.target.value == 'selectAll') { displayEpisodes(allEpisodes); }
+  for (let ep of allEpisodes) {
+    if (ep.id == selectEpisodesEl.value) {
+      episodesEl.innerHTML = getEpisodeCard(ep);
+      searchResult.innerHTML = `Displaying 1/${allEpisodes.length}`;
+    }
+  }
+})
+
+selectShowEl.addEventListener('change', (e) => {
+  if (e.target.value == 'allShows') {
+    displayShows(allTVShows);
+    searchBar(allTVShows);
```

Comment:
I can't find where this function defined.

Reply:
Thank you!
Much appreciated
I have cleaned the code a bit
Let me know what you think